src/Reflection/Native/NativeMethodReflection.php

```diff
@@ -33,6 +33,7 @@ public function __construct(
 		private Assertions $assertions,
 		private ?Type $selfOutType,
 		private ?string $phpDocComment,
+		private TrinaryLogic $returnsByReference,
```

Comment:
Instead of constructor, it should be added on BuiltinMethodReflection and read from there.